tests/core/nogo/staticcheck/staticcheck_test.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2019 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package staticcheck_test
+
+import (
+	"bytes"
+	"regexp"
+	"testing"
+
+	"github.com/bazelbuild/rules_go/go/tools/bazel_testing"
+)
+
+func TestMain(m *testing.M) {
+	bazel_testing.TestMain(m, bazel_testing.Args{
+		Main: `
+-- BUILD.bazel --
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_tool_library", "nogo")
+load("@io_bazel_rules_go//go/analyzer/staticcheck:def.bzl", "staticcheck_analyzer")
+
+nogo(
+    name = "nogo",
+		deps = staticcheck_analyzer(["ST1000"]),
+    visibility = ["//visibility:public"],
+)
+
```

Comment:
FYI: This block shows the UX I am aiming for:

`staticcheck_analyzer` being a macro that return a list of analyzers that can be used in `nogo` deps attribute.

This being a list also means that folks can compose it freely in the future. Example would be:

```
nogo(
    name = "nogo",
		deps = TOOLS_NOGO + staticcheck_analyzer(STATICCHECK_ALL_ANALYZER) + golangci_lint_analyzer(["gofumpt"]),
    visibility = ["//visibility:public"],
)
```